sources/client/representation/qt/gaminglineedit.cpp

```diff
@@ -28,8 +26,32 @@ void GamingLineEdit::keyPressEvent(QKeyEvent *event)
     }
     if (event->key() == Qt::Key_F2)
     {
-        setText("OOC ");
+        ToggleOOC();
         return;
     }
     QLineEdit::keyPressEvent(event);
 }
+void GamingLineEdit::ToggleOOC(){
+    QString current_text = text();
```

Comment:
`current_text` isn't changed later, so it would be wise to mark it as `const` to avoid mistakes.
```
const QString current_text = text();
```